README.md
See the [mod page](https://mods.factorio.com/mod/Pre0-17-60Oil) on the
Factorio mod site for a description of what this is.


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.2.1
Date: 2021/11/22
  Changes:
    - Simple changes to displayed name and descriptions to make more sense for newer players that never played Factorio prior to v0.17.60
---------------------------------------------------------------------------------------------------
Version: 0.2.0
Date: 2021/02/03
  Changes:
    - Update for 1.1.
---------------------------------------------------------------------------------------------------
Version: 0.1.2
Date: 2020/03/17
  Changes:
    - Add Changelog entries for changes.
---------------------------------------------------------------------------------------------------
Version: 0.1.1
Date: 2020/03/17
  Changes:
    - Stop modifying logistics slot and autotrash slot researches that
      no longer exist in Factorio 0.18.13
---------------------------------------------------------------------------------------------------
Version: 0.1.0
Date: 2020/01/31
  Changes:
    - Updated Factorio version to 0.18. No other changes seem necessary.
---------------------------------------------------------------------------------------------------
Version: 0.0.5
Date: 2019/08/05
  Changes:
    - Added options to configure which parts are reverted back to pre 0.17.60
  Bugfixes:
    - Fixed a bug with reverting the lubricant tech prereq
---------------------------------------------------------------------------------------------------
Version: 0.0.4
Date: 2019/07/30
  Changes:
    - Updated info.json file to match descriptions and add forum thread link
---------------------------------------------------------------------------------------------------
Version: 0.0.3
Date: 2019/07/30
  Changes:
    - Updated entries in version 0.0.1 & 0.0.2 change log
    - Technologies
        - Chemical Science Pack Prerequisites: removed "sulfur-processing"
        - Laser Prerequisites: removed "chemical-science-pack"
        - Laser Science Packs: removed "chemical-science-pack"
        - Rocketry Prerequisites: replaced "flammables" with "rocket-fuel"
        - Worker Robots Speed 1 Prerequisites: added "chemical-science-pack"
        - Worker Robots Storage 1 Prerequisites: added "chemical-science-pack"
        - Auto Character Logistic Trash Slots Prerequisites: added "chemical-science-pack"
        - Personal Roboport Equipment Prerequisites: added "chemical-science-pack"
    - Recipes
        - Chemical Science Pack Ingredients: replaced sulfur with solid fuel
        - Rocket Fuel Category: replaced "crafting-with-fluid" with "crafting"
---------------------------------------------------------------------------------------------------
Version: 0.0.2
Date: 2019/07/30
  Changes:
    - Technologies
        - Rocket Fuel Prerequisites: removed "advanced-oil-processing"
        - Rocket Fuel Science Packs: removed "chemical-science-pack"
    - Recipes
        - Rocket Fuel Ingredients: removed light oil
        - Flamethrower Ammo Ingredients: replaced 100x crude oil with 50x each heavy & light oil
---------------------------------------------------------------------------------------------------
Version: 0.0.1
Date: 2019/07/30
  Features:
    - Initial release
  Changes:
    - Technologies
        - Laser Turrets Science Pack Costs: 150 -> 200
        - Laser Turrets Science Packs: removed "chemical-science-pack"
        - Robotics Science Pack Costs: 75 -> 150
        - Robotics Science Packs: removed "chemical-science-pack"
        - Lubricant Prerequisites: replaced "advanced-oil-processing" with "oil-processing"
        - Lubricant Science Packs: removed "chemical-science-pack"
        - Electric Engine Science Pack Costs: 50 -> 100
        - Electric Engine Science Packs: removed "chemical-science-pack"
        - Construction Robotics Prerequisites: added "advanced-electronics"
        - Construction Robotics Science Packs: removed "chemical-science-pack"
        - Logistic Robotics Prerequisites: added "advanced-electronics"
        - Logistic Robotics Science Packs: removed "chemical-science-pack"
        - Character Logistic Slots 1 Science Pack Costs: 50 -> 100
        - Character Logistic Slots 1 Science Packs: removed "chemical-science-pack"
        - Character Logistic Slots 2 Science Pack Costs: 75 -> 150
        - Character Logistic Slots 2 Science Packs: removed "chemical-science-pack"
        - Character Logistic Trash Slots 1 Science Pack Costs: 30 -> 100
        - Character Logistic Trash Slots 1 Science Packs: removed "chemical-science-pack"
        - Oil Processing Recipe Unlocks: added solid fuel from light oil & heavy oil
        - Advanced Oil Processing Recipe Unlocks: removed solid fuel from light oil & heavy oil
    - Recipes
        - Basic Oil Processing Results: heavy oil 0 -> 30, light oil 0 -> 30, & petroleum gas 45 -> 40
        - Advanced Oil Processing Results: heavy oil 25 -> 10


settings.lua
data:extend({
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-chemscipack",
		description = "pre0-17-60oil-revert-chemscipack",
		setting_type = "startup",
		default_value = true,
		order = "a-a"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-lasertechs",
		description = "pre0-17-60oil-revert-lasertechs",
		setting_type = "startup",
		default_value = true,
		order = "a-b"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-rocketrytech",
		description = "pre0-17-60oil-revert-rocketrytech",
		setting_type = "startup",
		default_value = true,
		order = "a-c"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-roboticstechs",
		description = "pre0-17-60oil-revert-roboticstechs",
		setting_type = "startup",
		default_value = true,
		order = "a-d"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-solidfuel",
		description = "pre0-17-60oil-revert-solidfuel",
		setting_type = "startup",
		default_value = true,
		order = "b-b"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-rocketfuelrecipe",
		description = "pre0-17-60oil-revert-rocketfuelrecipe",
		setting_type = "startup",
		default_value = true,
		order = "b-c"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-flamethrowerammo",
		description = "pre0-17-60oil-revert-flamethrowerammo",
		setting_type = "startup",
		default_value = true,
		order = "b-d"
	},
	{
		type = "bool-setting",
		name = "pre0-17-60oil-revert-advoilrecipe",
		description = "pre0-17-60oil-revert-advoilrecipe",
		setting_type = "startup",
		default_value = true,
		order = "b-a"
	}
})

prototypes/recipes.lua
local basicoilprocessingrecipe = data.raw["recipe"]["basic-oil-processing"]
for _, result in pairs (basicoilprocessingrecipe.results) do
	if result.name == "petroleum-gas" then
		result.amount = 40
		break
	end
end
table.insert(basicoilprocessingrecipe.results, {type="fluid", name="heavy-oil", amount=30, fluidbox_index = 1})
table.insert(basicoilprocessingrecipe.results, {type="fluid", name="light-oil", amount=30, fluidbox_index = 2})

if settings.startup["pre0-17-60oil-revert-flamethrowerammo"].value then
	local flamethrowerammorecipe = data.raw["recipe"]["flamethrower-ammo"]
	for _, ingredient in pairs (flamethrowerammorecipe.ingredients) do
		if ingredient.name == "crude-oil" then
			ingredient.name = "light-oil"
			ingredient.amount = 50
			break
		end
	end
	table.insert(flamethrowerammorecipe.ingredients, {type="fluid", name="heavy-oil", amount=50})
end

if settings.startup["pre0-17-60oil-revert-advoilrecipe"].value then
	local advancedoilprocessingrecipe = data.raw["recipe"]["advanced-oil-processing"]
	for _, result in pairs (advancedoilprocessingrecipe.results) do
		if result.name == "heavy-oil" then
			result.amount = 10
			break
		end
	end
end

if settings.startup["pre0-17-60oil-revert-chemscipack"].value then
	local chemicalsciencepackrecipe = data.raw["recipe"]["chemical-science-pack"]
	for _, ingredient in pairs (chemicalsciencepackrecipe.ingredients) do
		if ingredient[1] == "sulfur" then
			ingredient[1] = "solid-fuel"
			break
		end
	end
end

if settings.startup["pre0-17-60oil-revert-rocketfuelrecipe"].value then
	local rocketfuelrecipe = data.raw["recipe"]["rocket-fuel"]
	rocketfuelrecipe.category = "crafting"
	for p, ingredient in pairs (rocketfuelrecipe.ingredients) do
		if ingredient.name == "light-oil" then
			rocketfuelrecipe.ingredients[p] = nil
			break
		end
	end
end

prototypes/technologies.lua
if settings.startup["pre0-17-60oil-revert-solidfuel"].value then
	local oilprocessingtech = data.raw["technology"]["oil-processing"]
	table.insert(oilprocessingtech.effects, {type = "unlock-recipe", recipe = "solid-fuel-from-light-oil"})
	table.insert(oilprocessingtech.effects, {type = "unlock-recipe", recipe = "solid-fuel-from-heavy-oil"})

	local advancedoilprocessingtech = data.raw["technology"]["advanced-oil-processing"]
	for p, effect in pairs (advancedoilprocessingtech.effects) do
		if effect.recipe == "solid-fuel-from-light-oil" or effect.recipe == "solid-fuel-from-heavy-oil" then
			advancedoilprocessingtech.effects[p] = nil
		end
	end
end

if settings.startup["pre0-17-60oil-revert-chemscipack"].value then
	local chemicalsciencepack = data.raw["technology"]["chemical-science-pack"]
	for p, prereq in pairs (chemicalsciencepack.prerequisites) do
		if prereq == "sulfur-processing" then
			chemicalsciencepack.prerequisites[p] = nil
			break
		end
	end
end

if settings.startup["pre0-17-60oil-revert-lasertechs"].value then
	local lasertech = data.raw["technology"]["laser"]
	for p, prereq in pairs (lasertech.prerequisites) do
		if prereq == "chemical-science-pack" then
			lasertech.prerequisites[p] = nil
			break
		end
	end
	lasertech.unit.count = 150
	for p, ingredient in pairs (lasertech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			lasertech.unit.ingredients[p] = nil
			break
		end
	end

	local laserturretstech = data.raw["technology"]["laser-turret"]
	laserturretstech.unit.count = 200
	for p, ingredient in pairs (laserturretstech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			laserturretstech.unit.ingredients[p] = nil
			break
		end
	end
end

if settings.startup["pre0-17-60oil-revert-rocketrytech"].value then
	local rocketfueltech = data.raw["technology"]["rocket-fuel"]
	for p, prereq in pairs (rocketfueltech.prerequisites) do
		if prereq == "advanced-oil-processing" then
			rocketfueltech.prerequisites[p] = nil
			break
		end
	end
	for p, ingredient in pairs (rocketfueltech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			rocketfueltech.unit.ingredients[p] = nil
			break
		end
	end

	local rocketrytech = data.raw["technology"]["rocketry"]
	for p, prereq in pairs (rocketrytech.prerequisites) do
		if prereq == "flammables" then
			rocketrytech.prerequisites[p] = "rocket-fuel"
			break
		end
	end
end

if settings.startup["pre0-17-60oil-revert-roboticstechs"].value then
	local roboticstech = data.raw["technology"]["robotics"]
	roboticstech.unit.count = 150
	for p, ingredient in pairs (roboticstech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			roboticstech.unit.ingredients[p] = nil
			break
		end
	end

	local lubricanttech = data.raw["technology"]["lubricant"]
	for p, prereq in pairs (lubricanttech.prerequisites) do
		if prereq == "advanced-oil-processing" then
			lubricanttech.prerequisites[p] = "oil-processing"
			break
		end
	end
	for p, ingredient in pairs (lubricanttech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			lubricanttech.unit.ingredients[p] = nil
			break
		end
	end

	local electricenginetech = data.raw["technology"]["electric-engine"]
	electricenginetech.unit.count = 100
	for p, ingredient in pairs (electricenginetech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			electricenginetech.unit.ingredients[p] = nil
			break
		end
	end

	local constructionroboticstech = data.raw["technology"]["construction-robotics"]
	table.insert(constructionroboticstech.prerequisites, "advanced-electronics")
	for p, ingredient in pairs (constructionroboticstech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			constructionroboticstech.unit.ingredients[p] = nil
			break
		end
	end

	local logisticroboticstech = data.raw["technology"]["logistic-robotics"]
	table.insert(logisticroboticstech.prerequisites, "advanced-electronics")
	for p, ingredient in pairs (logisticroboticstech.unit.ingredients) do
		if ingredient[1] == "chemical-science-pack" then
			logisticroboticstech.unit.ingredients[p] = nil
			break
		end
	end

	local workerrobotsspeed1tech = data.raw["technology"]["worker-robots-speed-1"]
	table.insert(workerrobotsspeed1tech.prerequisites, "chemical-science-pack")

	local workerrobotsstorage1tech = data.raw["technology"]["worker-robots-storage-1"]
	table.insert(workerrobotsstorage1tech.prerequisites, "chemical-science-pack")

	local personalroboportequipmenttech = data.raw["technology"]["personal-roboport-equipment"]
	table.insert(personalroboportequipmenttech.prerequisites, "chemical-science-pack")
end
